src/guide/receive_onchain.md

```diff
@@ -42,7 +47,12 @@ sdk.refund(refundable.bitcoin_address, destination_address, sat_per_vbyte).await
 
 ```swift
 do {
-  let swapInfo = try sdk.receiveOnchain()
+  // Optional user-selected dynamic fees
+  let lsp_info = try sdk.lsp_info();
+  let opening_fee_params = lsp_info?.opening_fee_params_list.values.first();
```

Comment:
Shouldn't we get the last value here (last expired)?
Also we need some explanations of the new fees structure and fields. We also need to mention the sorting and link to the lsp spec I think.
One option is to add the fee explanation as part of the "Connecting to the lsp" section: https://sdk-doc.breez.technology/guide/connecting_lsp.html

Reply:
I included a section introducing the dynamic fees, including link to spec and key points for using them.

In the example, it's hard to tell which fees the user would want, so for simplicity I kept the first and expanded the snippet comments saying this is the cheapest fee.

The intro section on the "Connecting to the LSP" page also describes sorting and expiration, as well as an empty opening fees arg.

Reply:
@JssDWt please let me know if the intro sounds good or if it needs more refining.

Reply:
From experience people are just copying and pasting the examples so I think the examples should target the common case and not the edge case.
In the common case people would not pass the fee params at all expecting the sdk to pick the default one.

Reply:
I think _if_ we add a section on fees, we should give a recommendation on how to pick the fees. And be clear about the fact that the SDK already has sane defaults for picking a fee.

Generally you'd want to pick a minimum time necessary to complete the payment. Then choose the first record in the list that satisfies the time constraint, because that is the cheapest option for that time constraint.

Reply:
That makes sense in receive payment over lightning but for receive on chain it will not work since the cheapest means the shortest time. In most cases the swap will take several hours which will cause the payment to fail (fee invalid anymore).
Also In receiveOnchain internally we pick the longer time, that's why this example is misleading and also will not work.

Reply:
Won't it be confusing if some snippets use first (cheapest) fees, when others use last (most expensive / biggest duration)?

Also, from a different perspective:  Assuming most people will copy-paste the snippet, the simplest would then be to have the snippets not use any custom fees, no? Snippets are small and simple, and docs are there for who wants to read more.

Reply:
Yes that's what I think. I agree with Jesse that it is not ideal that we explain the dynamic fees and not showing how to use it in a sample but the alternative of adding it in the basic samples invites more troubles then benefits.
IMO choosing manually a dynamic fee is a very advanced use case and most users will prefer letting the sdk choose for them.
@JssDWt what do you think of having in the future some kind of "Advanced" section where we can show these advanced scenarios? I really think that the basic examples should be kept simple.

Reply:
An Advanced section makes sense to me. Any optional parameters (like the receive_payment expiry & ctlv) could also be further explored there and keep the introduction to a minimal use case.

Reply:
> what do you think of having in the future some kind of "Advanced" section where we can show these advanced scenarios?
Yes, I agree this should be part of some advanced section.

> That makes sense in receive payment over lightning but for receive on chain it will not work since the cheapest means the shortest time.
That's what I meant! You first have to decide what the minimum time is you need (so the maximum time to get paid). For a swap we chose 48h. So your best fit for a fee params is the cheapest one that is valid for at least 48h. Same should be true for the 'normal invoice' actually. If we decide the minimum time is 15m (so you have max 15 minutes to get paid), you should choose the cheapest opening params with a validity of at least 15 minutes.

Reply:
@ok300 the Ruben's python PR that is based on this is good to go IMO. Can be merged and then this PR can be refined.
We need to remove the usage of fee params here, same as in the child PR.